examples/ui-demo/src/hooks/useRecurringTransactions.ts

```diff
@@ -87,22 +90,69 @@ export const useRecurringTransactions = ({
     },
   });
 
-  const handleTransaction = async (transactionIndex: number) => {
-    setTransactions((prev) => {
-      const newState = [...prev];
-      newState[transactionIndex].state = "initiating";
-      if (transactionIndex + 1 < newState.length) {
-        newState[transactionIndex + 1].state = "next";
-      }
-      return newState;
+  const { setToast } = useToast();
+
+  const handleError = (error: Error) => {
+    console.error(error);
+    setCardStatus("initial");
+    setTransactions(initialTransactions);
+    setToast({
+      type: "error",
+      text: "Something went wrong. Please try again.",
+      open: true,
     });
+  };
 
-    if (!sessionKeyClient) {
-      console.error("no session key client");
-      setCardStatus("initial");
+  // Handle the ticking of the timers.
+  useEffect(() => {
```

Comment:
related question re: timestamp and interval - could the reveal be handled in the 10s countdown promise? Not super high pri, just curious if we could simplify

Reply:
Not sure what you mean. Do you mean it would be ideal for each "complete" state to not fire until after the 10 sec has passed?

Reply:
Hmm, we could probably just give each of them the time they should fire at the beginning and set the timeouts to 0, 10, and 20 sec right away. But doing it how it is now also results in waiting extra time if a txn takes longer than 10 sec, which I think is nice to have.